src/win/process.c

```diff
@@ -364,7 +365,7 @@ static WCHAR* search_path(const WCHAR *file,
 
   /* Check if the filename includes an extension */
   dot = wcschr(file_name_start, L'.');
-  name_has_ext = (dot != NULL && dot[1] != L'\0');
+  name_has_ext = (allow_no_extension || (dot != NULL && dot[1] != L'\0'));
```

Comment:
I have the impression this should just be hardcoded to 1 and removed, as a bugfix. The documentation in this file at least seems to explicitly state this check should not have been here

Reply:
> The documentation in this file at least seems to explicitly state this check should not have been here

Where is that stated? I see:

```
 * - If filename specified has *any* extension, search for the file with the
 *   specified extension first.
```

but nothing about files that have no extension.

To clarify, this is not about finding `exe.exe` when `exe` is requested, but about finding a file literally named `exe` with no extension when it exists and is requested.

Reply:
> If the literal filename is not found in a directory

It doesn't check for the literal filename (even if file_has_dir ) when this is false.

Reply:
OK, I'll make it check the filename unconditionally and update the documentation to reflect this.